modules/component-store/src/component-store.ts

```diff
@@ -149,8 +169,18 @@ export class ComponentStore<T extends object> implements OnDestroy {
    */
   private initState(state: T): void {
     scheduled([state], queueScheduler).subscribe((s) => {
+      const isInitialized = this.isInitialized;
       this.isInitialized = true;
       this.stateSubject$.next(s);
+
+      if (!isInitialized) {
+        const onStateInit: Function | undefined = (
+          this as unknown as ComponentStore<T> & OnStateInit
+        )['ngrxOnStateInit'];
+        if (typeof onStateInit === 'function') {
+          onStateInit.call(this, s);
+        }
+      }
```

Comment:
To avoid explicit casting here, I suggest this:

```suggestion
      if (!isInitialized && isOnStateInitDefined(this)) {
        this.ngrxOnStateInit();
      }
```

```ts
function isOnStateInitDefined(cs: unknown): cs is OnStateInit {
  return typeof (cs as OnStateInit).ngrxOnStateInit === 'function';
}
```

The same can be applied for `OnStoreInit` hook

Reply:
+1